builtin/credential/aws/path_role.go

```diff
@@ -20,6 +20,12 @@ func pathRole(b *backend) *framework.Path {
 				Type:        framework.TypeString,
 				Description: "Name of the role.",
 			},
+			"allowed_auth_types": {
```

Comment:
Good call out on the backwards compatibility, I think I've added the right things to preserve that (but please double check anywhere else I might have missed!).

However, I think I might be saying something different than you are here. I'm talking about preserving the backwards compatibility of scripts that do things like:

```bash
vault auth-enable aws-ec2
vault write auth/aws-ec2/role/myrole bound_ami_id=ami-xxxxxxx
```

I'm trying to avoid changing the behavior of scripts like those that aren't aware of the IAM auth method. If we make the default `ec2,iam` and further don't default to inferencing, then the write I just mentioned will fail because there's no binding to something the `iam` auth_method can understand.

Again, I'm mostly making assumptions about what sort of backwards compatibility you want to support and trying to fit that. If you're OK changing that level of backwards compatibility given the implications, then I'm happy to change the code to support that.